components/TrainList.jsx

```diff
@@ -0,0 +1,54 @@
+import React, { Component } from 'react';
+import Link from 'next/link';
+// import TrainDetails from './TrainDetails';
+
+class TrainList extends Component {
+  state = {
```

Comment:
You need to use the proper constructor for a component
```
constructor(props) {
  super(props)
  this.state = {/* All your state */}
}
```
Still, I'm not sure all of that data belongs in state. State should generally only be used for data that will change state.